c10/core/CachingDeviceAllocator.h

```diff
@@ -645,4 +651,94 @@ ExpandableSegmentT* make_expandable_segment(
   return ptr;
 }
 
+typedef std::shared_ptr<GatheredContext> (*CreateContextFn)();
```

Comment:
Nit: cleaner as using, right? Why typedef?

Reply:
Good catch! This originally moved from: https://github.com/pytorch/pytorch/blob/842cc77ab9aafd518593c2fce077d6abb42a5b7f/c10/cuda/CUDACachingAllocator.h#L56 I will change it to use `using` keyword.